backend/routes/post.js

```diff
@@ -53,6 +53,29 @@ router.get('/', (request, response, next) => {
 router.get('/:_pid', async (request, response, next) => {
   let body = request.post.toObject()
   const user = await User.getUser(request, response, next, request.post.author)
+  for (let i = 0; i < body.comments.length; i++) {
+    const comment = await Comment.getComment(
+      request,
+      response,
+      next,
+      body.comments[i]
+    )
+    body.comments[i] = comment.toObject()
+    const author = await User.getUser(
+      request,
+      response,
+      next,
+      body.comments[i].author
+    )
+    body.comments[i].authorName = author.name
+    // body.comments[i].author = await User.getUser(
+    //   request,
+    //   response,
+    //   next,
+    //   body.comments[i].author
+    // )
+    // body.comments[i].author = body.comments[i].author.name
+  }
```

Comment:
These are awesome codes, although I think some way to make it easier to read / better if we follow the modularization pattern of MVC where we put functions that manipulate the model inside the controller, and we use the routes handle calling controllers, and handling errors prior to calling the controller.

In that case, we separate the logic of data manipulation and interfacing of the backend to the frontend/requester.

Reply:
I'm not sure what you mean. I'll read up on MVC to learn about it though.

Reply:
The first time I observed that concept was [here](https://github.com/JeremiahPinto/HomeForNow/tree/master/server) 

here's the converstation related to that
![image](https://user-images.githubusercontent.com/44391389/74945000-f3e19780-5431-11ea-989c-b157c52657f0.png)

Going back to what you said about the `[Schema].statics.[methods]`, I don't know how the code organization apply to that, but that seems very neater than what I was trying to say